src/charm.py

```diff
@@ -57,44 +67,90 @@ def _on_install(self, _):
         """Handler for the install event during which we will update the K8s service."""
         self._patch_k8s_service()
 
-    def _on_config_changed(self, _):
-        if self._stored.provider_ready:
-            self.unit.status = ActiveStatus()
+    def _on_config_changed(self, event):
+        self._configure(event)
 
     def _on_loki_pebble_ready(self, event):
-        """Define and start a workload using the Pebble API."""
-        # Get a reference the container attribute on the PebbleReadyEvent
-        container = event.workload
-        # Define an initial Pebble layer configuration
+        self._configure(event)
+
+    def _on_alertmanager_change(self, event):
+        self._configure(event)
+
```

Comment:
on upgrade should probably also call _configure, and possibly manipulate self._stored.config as well.

Reply:
See #13